app/services/scorecardService.js

```diff
@@ -190,6 +199,56 @@ const scorecardService = (() => {
 
     return data;
   }
+
+  // ------------------Removing Scorecard--------------------
+
+  function removeScorecardAsset(uuid, callback) {
+    let scorecard = realm.objects('Scorecard').filtered(`uuid='${uuid}'`)[0];
+    let tables = ['CustomIndicator', 'Facilitator', 'Participant', 'ProposedCriteria', 'VotingCriteria', 'Rating'];
+
+    realm.write(() => {
+      for(let i=0; i<tables.length; i++) {
+        let collection = realm.objects(tables[i]).filtered(`scorecard_uuid='${uuid}'`);
```

Comment:
@hengsokly , just curious if we could do a kind like OOP then load the model object, and every model has to check for hasDependency() -> model.removeDependency() if it has